mountpoint-s3/src/cli.rs

```diff
@@ -918,6 +922,15 @@ fn validate_mount_point(path: impl AsRef<Path>) -> anyhow::Result<()> {
     Ok(())
 }
 
+#[cfg(feature = "sse_kms")]
+fn validate_sse_args(sse_type: &Option<String>, sse_kms_key_id: &Option<String>) -> anyhow::Result<()> {
+    if sse_kms_key_id.is_some() && *sse_type == Some("AES256".to_owned()) {
```

Comment:
Could you rewrite this check as a pattern match?

Reply:
the `if` is clearer for my eye, but may be I am just not used to this style:
```
match sse_kms_key_id {
    Some(_) => {
        match sse_type {
            Some("AES256") => Err(anyhow!("--sse-kms-key-id can not be used with --sse AES256")),
            _ => Ok(())
        }
    }
    _ => Ok(())
}
```

Reply:
Forgive me if I'm wrong, I think Ale was thinking something more along the lines of...

```rust
match (sse_type, sse_kms_key_id) {
    (Some("AES256"), Some(_key_id)) => Err(...)
    ...
}
```

Reply:
Using my timezone advantage to merge this before you can bikeshed it :-)

Reply:
For posterity, now that we can no longer bikeshed: my preference would have been for @dannycjones' suggestion, but after the change to `&str` (and removal of `.to_owned()`) it would have been a minor nit.